programs/create-genesis/main.cpp

```diff
@@ -124,11 +141,25 @@ void read_contract(const genesis_info::account& acc, contract_abicode& abicode)
     };
 }
 
+void config_reader::read_options(const variables_map& options) {
+    if (!options["state"].as<bfs::path>().empty()) {
+        create_genesis = true;
```

Comment:
it should always be true. it's light enough and makes code simpler, comparing with optional execution